### PR TITLE
PIMS-653/645 MUI Setup and Landing Page

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

[PIMS-653](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-653)
[PIMS-645](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-645)

* Installed latest version of MUI and BC Government fonts.
* Added assets for BC gov logo and home page vector graphic.
* Added a Header and Footer component, as well as a Home/Landing page.
* Made a BaseLayout component we can reuse for different pages that will wrap any children in the header and footer (optional).
* Wrapped the app in a global MUI theme, contains a palette override, typography settings, and tweaks for the buttons and links.
## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: react-app/package.json [hunk 1]
```diff
@@ -12,7 +12,11 @@
     "check": "prettier --check \"./src/**/*.{js,jsx,ts,tsx,css,scss}\""
   },
   "dependencies": {
+    "@bcgov/bc-sans": "^2.1.0",
     "@bcgov/citz-imb-kc-react": "https://github.com/bcgov/citz-imb-kc-react/releases/download/v1.1.0/bcgov-citz-imb-kc-react-1.1.0.tgz",
+    "@emotion/react": "^11.11.3",
+    "@emotion/styled": "^11.11.0",
+    "@mui/material": "^5.15.3",
```
Comment (dbarkowsky) [inline react-app/package.json:19]: Recommend we lock down these versions.
```suggestion
    "@bcgov/bc-sans": "2.1.0",
    "@bcgov/citz-imb-kc-react": "https://github.com/bcgov/citz-imb-kc-react/releases/download/v1.1.0/bcgov-citz-imb-kc-react-1.1.0.tgz",
    "@emotion/react": "11.11.3",
    "@emotion/styled": "11.11.0",
    "@mui/material": "5.15.3",
```

File: react-app/src/components/layout/BaseLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React, { PropsWithChildren } from 'react';
+import { Box } from '@mui/material';
+import Header from '@/components/layout/Header';
+import Footer from '@/components/layout/Footer';
+
+interface IBaseLayoutProps extends PropsWithChildren {
```
Comment (dbarkowsky) [inline react-app/src/components/layout/BaseLayout.tsx:6]: > PropsWithChildren

I didn't know this existed, but super helpful.

File: react-app/src/components/layout/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        <Box>
+          <picture>
+            <source srcSet={headerImageLarge} media="(min-width: 960px)"></source>
+            <source srcSet={headerImageSmall} media="(min-width: 600px)"></source>
+            <img
+              src={headerImageSmall}
+              alt={'Government of British Columbia'}
+              style={{ height: '48px' }}
+            />
+          </picture>
```
Comment (dbarkowsky) [inline react-app/src/components/layout/Header.tsx:34]: This is also really cool. Wish I knew about `<picture>` sooner.

File: react-app/src/components/layout/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        </Box>
+        <Box>
+          <Divider
+            style={{ borderBottomWidth: '26px', borderRightWidth: '2px' }}
+            orientation="vertical"
+          />
+        </Box>
```
Comment (dbarkowsky) [inline react-app/src/components/layout/Header.tsx:41]: Does a Divider need a Box if it's the only child?

**Review by dbarkowsky — APPROVED** [review]
Looks great. Really like the theme provider setup. 
Going to approve, but would like to have a talk as a group about how we want to organize styling sometime in the future. I'll set something up for next week.
Mostly thinking about: 
- how much inline styling we want vs when to separate
- when we're using em vs px, rem, vh, etc.

File: react-app/src/components/layout/Header.tsx (hunk 4 above)
Comment (GrahamS-Quartech) [reply react-app/src/components/layout/Header.tsx]: It doesn't, not sure why I put those there. Removed.

**Pull request merged**